### ISSUE TITLE
Tests

### ISSUE THREAD

**Issue by Infernio** [wrye-bash/wrye-bash#508]
We will want to have tests for all of Wrye Bash eventually, this is the long-term tracking issue.

## TL;DR
- High Priority:
  - `patcher`
  - `brec`
  - `ini_files.py`
  - `load_order.py`
  - `mod_files.py`
  - `record_groups.py`
- Easy:
  - `bosh`:
    - `loot_parser.py` (Designed with testing in mind)
    - `loot_conditions.py` (Designed with testing in mind)
  - `fomod.py` (Had tests sitting around, might be too low-level though)

## Full Overview
- `basher`: None yet
- `bosh`: None except in the following files:
  - `cosaves.py`: Pluggy tests cases missing, LMOD and LIMD test cases missing (ff36f7359895309e8326550c66a7da947686f105)
- `brec`: None yet, high priority
- `game`: None yet, not sure if needed
- `gui`: None yet
- `patcher`: None yet, high priority
- `tests`: Not needed, duh
- `__init__.py`: Empty
- `_games_lo.py`: None yet, temp-ish module
- `archives.py`: None yet
- `balt.py`: Deprecated module
- `barb.py`: None yet
- `barg.py`: None yet
- `bash.py`: None yet
- `bass.py`: Not needed
- `belt.py`: None yet
- `bolt.py`: `LowerDict` and subclasses (f5199336a0ade98cdee5346547e3165df35a638b), decode/encode/getbestencoding (4c1bc4e1220bdc76bf49244e478fafb08f59aeb8)
- `bush.py`: None yet
- `cint.py`: None yet, probably futile (probably only test as part of CBash patchers, don't want to waste effort on testing code that we might throw away on py3/rbash)
- `env.py`: None yet, needs rework anyways
- `exception.py`: Probably not needed? Could try raising each one I guess
- `fomod.py`: Ganda wrote some tests, will try to resurrect those or write new ones otherwise
- `ini_files.py`: None yet, high priority
- `initialization.py`: None yet
- `load_order.py`: None yet, high priority
- `localize.py`: None yet
- `mod_files.py`: None yet, high priority (closely ties in with `brec`)
- `parsers.py`: None yet, has a big outstanding refactoring so pointless right now
- `record_groups.py`: None yet, high priority (closely ties in with `brec`)
- `ScriptParser.py`: None yet, should probably wait for [wizparse](https://github.com/Infernio/wizparse) so we don't test some of the spec violations the current parser allows
- `windows.py`: None yet, see `env.py`; this might as well be `_windows.py`

**Comment by Utumno** [comment]
Trying to wrap my brain around https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure - src corresponds to mopy I guess. Something that we may need to keep in mind if ever we end up in PyPI

**Comment by Infernio** [comment]
Had a look at the fomod tests - some seem useful, others are probably too close to internals and would just break every time we refactor something. Plus they're ancient (still using `boop` :P), so would need rewriting anyways.
